rest-service/manager_rest/resource_manager.py

```diff
@@ -359,6 +359,23 @@ def remove_plugin(self, plugin_id, force):
                                                      plugin.archive_name)
         shutil.rmtree(os.path.dirname(archive_path), ignore_errors=True)
 
+    def upload_blueprint(self, blueprint_id, app_file_name, blueprint_url,
+                         file_server_root):
+        try:
+            self._execute_system_workflow(
```

Comment:
So to recap, current solution is `execute_system_workflow` with is_system_workflow=False (see comment below) so that it won't block. Executing it through `execute_workflow` assumes having a deployment for it, which we don't have at this stage. 
In the future I suggest reworking such workflows ('create_deployment_environment, 'delete_deployment_environment', 'upload_blueprint') into a separate class, if only for clarity's sake